### PR TITLE
CHANGELOG: update changelog

### PR DESCRIPTION
**Pull request by squat**
This commit updates the changelog to reflect the new changes since release 0.22.2. Once this is merged, we can cut the new release branch and make the release PR with new versions.

cc @mxinden

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Upcoming
 
+* [FEATURE] Introduce new flag to control logging format (#1475)
+* [FEATURE] Ensure Prometheus Operator container runs as `nobody` user by default (#1393)
+* [BUGFIX] Fix reconciliation of ServiceMonitors and Rules when a single namespace is being watched (#1749)
```
Comment (mxinden) [inline CHANGELOG.md]: We are not touching (reconciling) ServiceMonitors or PrometheusRules. How about `Fix reconciliation of Prometheus StatefulSet due to ServiceMonitor and PrometheusRule when a ...`?

**Comment by mxinden** [comment]
Do you want to bump the version in a follow up PR, or directly here @squat?

**Comment by squat** [comment]
@mxinden will bump versions directly in the release branch in a specific PR

**Comment by mxinden** [comment]
@squat Just keep in mind, that that would need to be merged back into master then.

**Comment by squat** [comment]
@mxinden we need to be consistent with this. Our last two releases had version bumps directly in the release branch [0] [1], while the one before that bumped the versions in master [2]. It's not a big deal; we just need to pick a practice and stick with it. I'll bump versions directly in this PR, merge to master, and then cut the release branch from master. I suggest we stick with this consistent practice going forward.

[0] https://github.com/coreos/prometheus-operator/pull/1683
[1] https://github.com/coreos/prometheus-operator/pull/1642
[2] https://github.com/coreos/prometheus-operator/pull/1597

File: jsonnet/prometheus-operator/prometheus-operator.libsonnet [hunk 2]
```diff
@@ -119,8 +119,8 @@ local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
         container.withPorts(containerPort.newNamed('http', targetPort)) +
         container.withArgs([
           '--kubelet-service=kube-system/kubelet',
-          # Prometheus Operator is run with a read-only root file system. By
-          # default glog saves logfiles to /tmp. Make it log to stderr instead.
+          // Prometheus Operator is run with a read-only root file system. By
+          // default glog saves logfiles to /tmp. Make it log to stderr instead.
```
Comment (squat) [inline jsonnet/prometheus-operator/prometheus-operator.libsonnet]: Automatically fixed by vim-jsonnet plugin. Added in separate commit.

**Comment by squat** [comment]
seeing an expected error during manifest generation from jsonnet since the targeted tag does not exist yet

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
-# Upcoming
+## 0.23.0 / 2018-08-06
 
+* [FEATURE] Introduce new flag to control logging format (#1475)
+* [FEATURE] Ensure Prometheus Operator container runs as `nobody` user by default (#1393)
+* [BUGFIX] Fix reconciliation of Prometheus StatefulSets due to ServiceMonitors and PrometheusRules changes when a single namespace is being watched (#1749)
 * [CHANGE] Deprecate specification of Prometheus rules via ConfigMaps in favor of `PrometheusRule` CRDs
```
Comment (mxinden) [inline CHANGELOG.md]: Can you order the entries like:

```
Entries in the CHANGELOG.md are meant to be in this order:

    [CHANGE]
    [FEATURE]
    [ENHANCEMENT]
    [BUGFIX]
```

https://github.com/prometheus/prometheus/wiki/HOWTO-cut-a-new-release

**Comment by mxinden** [comment]
@squat In terms of consistency: Minor releases are bumped in `master`. Patch releases are bumped in the release branch. What do you think?

**Comment by squat** [comment]
@mxinden got it. that bumping makes sense. thanks!

File: contrib/kube-prometheus/jsonnet/kube-prometheus/jsonnetfile.json [hunk 4]
```diff
@@ -38,7 +38,7 @@
                     "subdir": "jsonnet/prometheus-operator"
                 }
             },
-            "version": "v0.22.2"
+            "version": "v0.23.0"
```
Comment (brancz) [inline contrib/kube-prometheus/jsonnet/kube-prometheus/jsonnetfile.json]: this should be reverted

File: bundle.yaml [hunk 5]
```diff
@@ -99,8 +99,8 @@ spec:
         - --kubelet-service=kube-system/kubelet
         - -logtostderr=true
         - --config-reloader-image=quay.io/coreos/configmap-reload:v0.0.1
-        - --prometheus-config-reloader=quay.io/coreos/prometheus-config-reloader:v0.22.2
-        image: quay.io/coreos/prometheus-operator:v0.22.2
+        - --prometheus-config-reloader=quay.io/coreos/prometheus-config-reloader:v0.22.1
```
Comment (brancz) [inline bundle.yaml]: these should be 0.23.0

Comment (squat) [reply bundle.yaml]: funny, this was the result of running make generate. make generate-in-docker produces very different results.

**Comment by mxinden** [comment]
@squat could we wait for https://github.com/coreos/prometheus-operator/pull/1753 before merging here?

**Pull request merged**